ws-client/src/client.rs

```diff
@@ -134,17 +154,18 @@ impl WsClient {
 	/// Initializes a new WebSocket client
 	///
 	/// Fails when the URL is invalid.
-	pub async fn new(remote_addr: impl AsRef<str>, config: WsConfig) -> Result<Self, Error> {
-		let (sender, receiver) = jsonrpc_transport::websocket_connection(remote_addr.as_ref())
-			.await
-			.map_err(|e| Error::TransportError(Box::new(e)))?;
+	pub async fn new(config: WsConfig<'_>) -> Result<WsClient, Error> {
+		let max_capacity_per_subscription = config.max_subscription_capacity;
+		let request_timeout = config.request_timeout;
+		let (to_back, from_front) = mpsc::channel(config.max_concurrent_requests_capacity);
 
-		let (to_back, from_front) = mpsc::channel(config.request_channel_capacity);
+		let (sender, receiver) =
+			jsonrpc_transport::websocket_connection(config).await.map_err(|e| Error::TransportError(Box::new(e)))?;
 
 		async_std::task::spawn(async move {
-			background_task(sender, receiver, from_front, config).await;
+			background_task(sender, receiver, from_front, max_capacity_per_subscription).await;
 		});
-		Ok(Self { to_back, config })
+		Ok(Self { to_back, request_timeout })
```

Comment:
removed `Config` from the client, ideally, everything should be handled by the `transport` but we have two exceptions: background thread and timeouts.